version: "3.7"

networks:
  dzangolab-public:
    external: true

services:
  app:
    deploy:
      labels:
        - "traefik.docker.network=dzangolab-public"
        - "traefik.enable=true"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-http.entrypoints=web"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-http.middlewares=https-redirect"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-http.rule=Host(`ujwalk-courses-app.sandbox.dzangohq.org`)"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-https.entrypoints=websecure"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-https.rule=Host(`ujwalk-courses-app.sandbox.dzangohq.org`)"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-https.tls.certresolver=le"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-https.tls.domains[0].main=ujwalk-courses-app.sandbox.dzangohq.org"
        - "traefik.http.routers.sandbox-ujwalk-courses-app-https.tls=true"
        - "traefik.http.services.sandbox-ujwalk-courses-app.loadbalancer.server.port=80"
      resources:
        limits:
          cpus: "0.50"
          memory: 500M
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
    image: 139549850843.dkr.ecr.ap-southeast-1.amazonaws.com/sandbox/courses-app:0.1.develop-ujwalk
    networks:
      - dzangolab-public
